.ci/cico_build_and_deploy_custom_rh-che.sh

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env bash
+# Copyright (c) 2018 Red Hat, Inc.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+
+set -x
+set -e
+set +o nounset
+
+/usr/sbin/setenforce 0
+
+export ORIGIN_CLIENTS_URL=http://mirror.centos.org/centos/7/paas/x86_64/openshift-origin/origin-clients-3.7.1-2.el7.x86_64.rpm
+export OPENSHIFT_RDU2C_URL=https://dev.rdu2c.fabric8.io:8443/
+export OC_VERSION=3.9.19
+
+function tagAndPushDocker() {
+  export RH_CHE_AUTOMATION_BUILD_TAG=rh-che-automated-build-$(git rev-parse --short HEAD)
+  export CHE_DOCKER_BASE_IMAGE=eclipse/che-server:nightly-centos
+  echo "che-docker-base-image:${CHE_DOCKER_BASE_IMAGE}"
+  docker pull ${CHE_DOCKER_BASE_IMAGE}
+  docker tag ${CHE_DOCKER_BASE_IMAGE} eclipse/che-server:local
+  rm -rf ./dockerfiles/che-fabric8/eclipse-che
+  mkdir ./dockerfiles/che-fabric8/eclipse-che
+  cp -r ./assembly/assembly-main/target/eclipse-che-fabric8-1.0.0-SNAPSHOT/eclipse-che-fabric8-1.0.0-SNAPSHOT/* ./dockerfiles/che-fabric8/eclipse-che/
+  bash ./dockerfiles/che-fabric8/build.sh || exit 1
+  docker rmi eclipse/che-server:local
+  docker rmi eclipse/che-server:nightly-centos
+  docker tag eclipse/che-server:nightly rhcheautomation/che-server:${RH_CHE_AUTOMATION_BUILD_TAG}
+  docker rmi eclipse/che-server:nightly
+  docker images
+  set +x
+  docker login -u ${RH_CHE_AUTOMATION_DOCKERHUB_USERNAME} -p ${RH_CHE_AUTOMATION_DOCKERHUB_PASSWORD}
+  set -x
+  docker push rhcheautomation/che-server:${RH_CHE_AUTOMATION_BUILD_TAG}
+}
+
+# Retrieve and test credentials
+
+if [ ! -f "./jenkins-env" ]; then
+  echo "CRITICAL ERROR: Jenkins env was not provided by jenkins job"
+  exit 1
+fi
+
+set +x
+cat ./jenkins-env | grep -E "(KEYCLOAK|BUILD_NUMBER|JOB_NAME|RH_CHE)" | sed 's/^/export /g' | sed 's/= /=/g' > ./export_env_variables
+if [ ! -f "./export_env_variables" ]; then
+  echo "CRITICAL ERROR: sed edit of ./jeninks_env failed"
+  exit 1
+fi
+
+source export_env_variables
+
+echo "Running ${JOB_NAME} build number #${BUILD_NUMBER}, testing creds:"
+
+CREDS_NOT_SET="false"
+#curl ${ORIGIN_CLIENTS_URL} > origin-clients.rpm
+#yum install --assumeyes ./origin-clients.rpm
+curl "https://mirror.openshift.com/pub/openshift-v3/clients/${OC_VERSION}/linux/oc.tar.gz" | tar xvz -C /usr/local/bin
+if [ -z ${RH_CHE_AUTOMATION_DOCKERHUB_USERNAME} ] || [ -z ${RH_CHE_AUTOMATION_DOCKERHUB_PASSWORD} ]; then
+  echo "Dockerhub credentials not set"
+  CREDS_NOT_SET="true"
```

Comment:
Why don't you just exit here instead of setting `CREDS_NOT_SET=true` and continuing? The same applies for the next `if` blocks. It's better to exit as soon as possible and avoiding the usage of the `CREDS_NOT_SET` env variable.

Reply:
Because I want to know which of the three secure store credentials fail.

Reply:
and echo an error message for each one.